source/pki_utils.c

```diff
@@ -226,3 +226,76 @@ int aws_read_and_decode_pem_file_to_buffer_list(
 
     return AWS_OP_SUCCESS;
 }
+
+int aws_decode_pem_to_der_buf(
+    struct aws_allocator *alloc,
+    const struct aws_byte_cursor *pem_cursor,
+    struct aws_byte_buf *der_buf) {
+    aws_byte_buf_clean_up(der_buf);
+    struct aws_array_list base_64_buffer_list;
+
+    if (aws_array_list_init_dynamic(&base_64_buffer_list, alloc, 2, sizeof(struct aws_byte_buf))) {
+        return AWS_OP_ERR;
+    }
+
+    int err_code = AWS_OP_ERR;
+
+    if (s_convert_pem_to_raw_base64(alloc, pem_cursor, &base_64_buffer_list)) {
+        goto cleanup_base64_buffer_list;
+    }
+
+    for (size_t i = 0; i < aws_array_list_length(&base_64_buffer_list); ++i) {

```

Comment:
in what cases would we get several base 64 buffers?

Reply:
We will get several base 64 buffers if there are more than one PEM data section. Updated the data so that we will try decoding and importing all data section into keychain until find the first succeed key.